### PR TITLE
fonttools: 4.21.1 -> 4.26.2;  psautohint: 2.3.0 -> 2.3.1;  afdko: 3.5.1 -> 3.7.1; nototools: 0.2.13 -> 0.2.16

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

Update a bunch of packages in the nototools font toolchain now that afdko 3.7.1 has been released for a bit (they took forever to release it and in between we were stuck with 3.5.x (holding other packages back) because it was broken on darwin).

Still outdated:

* fonttools (4.27.x isn't supported by afdko)
* ~~ufonormalizer (afdko's test suite doesn't like the behavior of 0.6.1)~~ (in `master` with workaround for the test suite)
* tqdm
* more?

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/afdko/default.nix [hunk 1]
```diff
@@ -2,37 +2,50 @@
...
   patches = [
-    # Skip date-dependent test. See
-    # https://github.com/adobe-type-tools/afdko/pull/1232
-    # https://github.com/NixOS/nixpkgs/pull/98158#issuecomment-704321117
-    (fetchpatch {
-      url = "https://github.com/adobe-type-tools/afdko/commit/2c36ad10f9d964759f643e8ed7b0972a27aa26bd.patch";
-      sha256 = "0p6a485mmzrbfldfbhgfghsypfiad3cabcw7qlw2rh993ivpnibf";
-    })
-    # fix tests for fonttools 4.21.1
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/afdko/default.nix:29]: Please remove the unused fetchpatch from inputs.

File: pkgs/development/python-modules/afdko/default.nix [hunk 2]
```diff
@@ -2,37 +2,50 @@
...
-    })
+    # Don't try to install cmake and ninja using pip
+    ./no-pypi-build-tools.patch
+    # Use antlr4 runtime from nixpkgs and link it dynamically
+    ./use-dynamic-system-antlr4-runtime.patch
   ];
 
-  nativeBuildInputs = [ setuptools-scm ];
+  preBuild = ''
+    ninja
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/afdko/default.nix:46]: calling ninja directly probably misses some arguments nix would add automatically.

File: pkgs/development/python-modules/afdko/no-pypi-build-tools.patch [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+index 50deb781..81417971 100644
+--- a/setup.py
++++ b/setup.py
+@@ -196,9 +196,7 @@ def main():
+           setup_requires=[
+               'wheel',
+               'setuptools_scm',
+-              'scikit-build',
+-              'cmake',
+-              'ninja'
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/afdko/no-pypi-build-tools.patch:20]: Please use subsituteInPlace instead.

File: pkgs/development/tools/parsing/antlr/4.9.nix [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      };
+    };
+  };
+
+  antlr = stdenv.mkDerivation {
+    pname = "antlr";
+    inherit version;
+
+    src = fetchurl {
+      url ="https://www.antlr.org/download/antlr-${version}-complete.jar";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/parsing/antlr/4.9.nix:54]:
```suggestion
      url = "https://www.antlr.org/download/antlr-${version}-complete.jar";
```

File: pkgs/development/tools/parsing/antlr/4.9.nix [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+          sha256 = "0nq7iajy9inllcspyqpxskfg3k5s1fwm7ph75i8lfc25rl35k1w7";
+        })
+      ];
+
+      patchFlags = [ "-p3" ];
+
+      nativeBuildInputs = [ cmake ninja pkg-config ];
+      buildInputs = [ utf8cpp ]
+        ++ lib.optional stdenv.isLinux libuuid
+        ++ lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.CoreFoundation;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/parsing/antlr/4.9.nix:34]:
```suggestion
        ++ lib.optional stdenv.isDarwin CoreFoundation;
```

File: pkgs/development/tools/parsing/antlr/4.9.nix [hunk 6]
```diff
@@ -0,0 +1,95 @@
+{ lib, stdenv, fetchurl, jre
+, fetchpatch, fetchFromGitHub, cmake, ninja, pkg-config, libuuid, utf8cpp, darwin }:
```
Comment (SuperSandro2000) [inline pkgs/development/tools/parsing/antlr/4.9.nix:2]:
```suggestion
, fetchpatch, fetchFromGitHub, cmake, ninja, pkg-config, libuuid, utf8cpp, CoreFoundation
}:
```

File: pkgs/development/python-modules/fonttools/default.nix [hunk 7]
```diff
@@ -14,37 +14,38 @@
...
     repo   = pname;
     rev    = version;
-    sha256 = "1x9qrg6ppqhm5214ymwvn0r34qdz8pqvyxd0sj7rkp06wa757z2i";
+    sha256 = "1zp9idjkn4bn1a4pn8x64vi8j1ijdsd4qvgf1f70dfwqvw6ak1i6";
   };
 
   # all dependencies are optional, but
   # we run the checks with them
   checkInputs = [
     pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fonttools/default.nix:39]:
```suggestion
```

File: pkgs/development/tools/parsing/antlr/4.9.nix (hunk 6 above)
Comment (sternenseemann) [reply pkgs/development/tools/parsing/antlr/4.9.nix:2]: That doesn't work?!

**Comment by sternenseemann** [comment]
`x86_64-darwin` is fine when disabling `utf8cpp`'s test suite, see <https://github.com/NixOS/nixpkgs/issues/144265>.

**Comment by sternenseemann** [comment]
Rebased on master (keeping staging target), simplified `afdko` derivation a bit.

Given that breaking change according to the release stabilization is

> A change that is likely to break downstream.

We may still merge this for 21.11, I think, since `afdko` has a relatively small set of (direct) reverse dependencies.

**Comment by sternenseemann** [comment]
Rebased on top of the last `staging-next` iteration and tested the font builds again. Gonna merge tomorrow, we could ponder backporting, but may be unnecessary hassle.

**Pull request merged**